backend/application/services/queue/consumer.py

```diff
@@ -6,85 +7,89 @@
 from application.services.tasks_definition.tasks import TASK_MAP
 from application.models.enums import TaskStatus
 from application.services.queue.callbacks import finished_callback
-from application.services.queue.models import Message, MessageKind
+from application.services.queue.models import Message
 from settings.default import MAX_TIMEOUT_SECONDS
 
 
+def do_check_task_status(task_id, task_record, task_listener, repository):
+    if task_record.status == TaskStatus.RUNNING:
+        if task_listener is None:
+            logging.warning(f"no listener for tasks {task_id} - skipping")
+            repository.mark_failed(obj=task_record)
+        else:
+            is_done = task_listener.done()
+            logging.info(
+                f"Task[{task_record.name}: {task_id}] -> It's done? R: {is_done}"
+            )
+            if is_done:
+                repository.mark_done(obj=task_record)
+            elif task_service.check_timeout(obj=task_record):
+                task_listener.cancel()
+                repository.mark_cancelled(obj=task_record)
+
+
+def clear_listeners():
+    """
+    if task_id in listeners:
+        del listeners[task_id]
+    """
+
+from application.services.tasks_definition.wait_5s import wait_5s
+
 async def consumer():
+    logging.info("LOL!!!!")
+    try:
+        cm = await TASK_MAP["wait_5s"](task_id="LOL!")
```

Comment:
what is this going on here? what are you trying to test?

Reply:
yes